### DISCUSSION TITLE
Always trigger CI/CD Workflow

### DISCUSSION THREAD

**Discussion by jonaswre** [microsoft/AL-Go#348] [Ideas]
If you want to enforce a successful build, you will need to set CI/CD Workflow to be required by the policy.

You will encounter some situations where no CI/CD Workflow is started, because no build is needed. 
In that case you will need to override the policy. 

I think it would be useful to start the CI/CD Workflow anyway and make it succeed immediately.

This adds more consistency in the process. 
Currently you need to know the common delay between PRHandler and CI/CD Workflow. If it takes longer then "normal" you need to understand if its not needed or something is wrong.

**Comment by jonaswre** [comment]
@freddydk have you encountered this?

**Reply by freddydk** [reply]
Yes, I know that the CI/CD trigger will start delayed and that I potentially can close the PR before the CI/CD process is done.
The CI/CD running as part of the PR is however only a test build and does not produce shippable artifacts
The Merge commit following the merge of the PR does produce a build and will produce artifacts.

Having said that, I don't know if there is a policy to always trigger CI/CD workflow? what is your suggestion?

**Reply by jonaswre** [reply]
No my suggestion is the PRHandler should always starts the CI/CD Workflow. The CI/CD Workflow should just immediately be successful if it doesn't need to do anything.

That way I can use the branch protection rule properly 

![image](https://user-images.githubusercontent.com/3731681/211761464-e8842e49-6771-43af-b2c5-e616c6f6f4f6.png)

It works fine if you start the CI/CD Workflow from the PRHandler. The only issue is if you don't start the workflow this will block the PR even if it "safe" to be merged.

**Reply by jonaswre** [reply]
I made this an Idea because I don't thing this a bug. I just think having proper support for the rule protection would allow you mitigate the problem that you can merge a PR after the PRHandler is successfull and before the CI/CD Workflow is spawned.

**Reply by freddydk** [reply]
Got it.
I don't really "start" the CI/CD workflow - it is kicked off by GitHub when the PRHandler completes.
I will think more about your scenario - it definitely has crossed my mind this issue.

**Reply by jonaswre** [reply]
I traced the "issue"
It was a PR that only consisted of .md files.

https://github.com/microsoft/AL-Go/blob/f4f2edb7fb17b7cc7f92b23210895405d33f053c/Templates/AppSource%20App/.github/workflows/PullRequestHandler.yaml#L6

why do you ignore them?

**Reply by freddydk** [reply]
The reason is really that there is nothing to build and it would just be a waste of resources - that's why I added the pattern back in the days.

**Reply by jonaswre** [reply]
I couldn't find anyway to exclude the rule if only md file are present.  Which would be a solution but this might require changes from github. 

Maybe the CICD Workflow could just decide if it needs to build.
That would increase the waste, but in relation to the whole build thats only small factor.

**Reply by freddydk** [reply]
Yeah, so always run the CI/CD, but complete quickly - sounds like the right thing to do

**Reply by jonaswre** [reply]
It would be good to be able to configure "complete quickly". Maybe there are other pattern that don't require a new build.

**Reply by freddydk** [reply]
With incremental builds - does this solve the issue?

**Reply by freddydk** [reply]
I actually think that this issue was created back when the CI/CD pipeline was also the pull request handler.
Today - the way to ensure this is by requiring Pull Requests and set the Pull Request Status Check as a status check, that needs to be successful:
![image](https://github.com/user-attachments/assets/19f9f24a-6600-4937-8e25-8cdc1b8a50c1)

In the PR build, this will look like:
![image](https://github.com/user-attachments/assets/3dc65f9b-ce53-40c2-9d24-ffcf8402d9b3)
